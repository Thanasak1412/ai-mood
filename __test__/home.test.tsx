import { toMatchDiffSnapshot } from "snapshot-diff";
import { vi } from "vitest";

import Home from "@/app/page";
import { User } from "@clerk/backend";
import { auth } from "@clerk/nextjs";
import { fireEvent, getByText, render } from "@testing-library/react";

vi.mock('@clerk/nextjs', () => ({
  auth: () => ({
    userId: 'user_2YVfHANK9tdUhFeXehtazTcDK5C',
  }),
}));

expect.extend({ toMatchDiffSnapshot });

describe('HomePage', () => {
  test('should render home correctly', () => {
    const { getByText } = render(<Home />);

    expect(getByText(/The best journal app, period/i)).toBeInTheDocument();
  });

  test('should redirect to journal entry page when is authenticated', () => {
    const { getByText, asFragment } = render(<Home />);

    const firstRender = asFragment();

    fireEvent.click(getByText(/Get started/i));

    expect(firstRender).not.toMatchSnapshot(asFragment());
  });
});
