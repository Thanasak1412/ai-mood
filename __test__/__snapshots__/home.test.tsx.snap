// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`__test__/home.test.tsx > HomePage > should render sign in page when is not authenticate 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <button>
-     Click to increase: 0
+     Click to increase: 1
    </button>
  </DocumentFragment>"
`;

exports[`HomePage > should redirect to journal entry page when is authenticated 1`] = `{}`;

exports[`HomePage > should render sign in page when is not authenticate 1`] = `{}`;
